File: backend/cloud-publishing-rest/employee-rest-service/src/main/java/com/ikuzmin/employee/rest/services/EmployeeService.java
package com.ikuzmin.employee.rest.services;

import com.ikuzmin.cloud.publishing.model.entitys.Employee;
import java.util.List;

/**
 *
 * @author Igor Kuzmin
 */
public interface EmployeeService {
  
  List<Employee> getAllEmployee();
  
  /*Optional<Employee> getEmployeeByLogin(String login);*/
  
}

File: backend/cloud-publishing-rest/employee-rest-service/src/main/kotlin/com/ikuzmin/employee/rest/services/EmployeeService.kt
package com.ikuzmin.employee.rest.services

import com.ikuzmin.cloud.publishing.model.dto.ProfileDto
import com.ikuzmin.cloud.publishing.model.dto.ShortProfileDto
import com.ikuzmin.cloud.publishing.model.entities.Education
import java.security.Principal


interface EmployeeService {

    fun getEmployeesProfiles(): List<ProfileDto>

    fun createEmployee(userProfile: ProfileDto)

    fun getEducationList(): List<Education>

    fun getUserPhoto(login: String): ByteArray?

    fun getShortProfile(principal: Principal): ShortProfileDto
}